ui/components/component-library/sensitive-text/sensitive-text.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Text } from '../text';
+import { SensitiveTextProps, SensitiveLengths } from './sensitive-text.types';
+
+export const SensitiveText = React.forwardRef<
+  HTMLParagraphElement,
+  SensitiveTextProps
+>((props, ref) => {
+  const {
+    isHidden,
+    length = SensitiveLengths.Short,
+    children,
+    ...restProps
+  } = props;
+  const fallback = '*'.repeat(length);
```

Comment:
memoize the fallback string to optimise performance for long lengths
 `const fallback = useMemo(() => '*'.repeat(length), [length]);`